draft/0006-pat.md

```diff
@@ -0,0 +1,127 @@
+3. Long-running processes that outlast typical session durations.
+
+## Personal Access Token (PAT) Creation
+
+The client makes a post request to the tenant's Management API `/api/users/:userId/personal-access-tokens` to create a PAT for the user. The following parameters are included in the HTTP request entity-body using the `application/json` format.
+
+1. `name`: REQUIRED. A user-defined name for the PAT to identify its purpose, should be unique in the user's account.
+2. `expiresAt`: OPTIONAL. The epoch time in milliseconds when the token will expire. If not provided, the token will never expire.
+
+If the request to create a PAT is successful, the tenant's Management API returns the PAT in the response.
+
+1. `name`: REQUIRED. The name of the PAT.
+2. `value`: REQUIRED. The value of the PAT.
+3. `expiresAt`: OPTIONAL. The epoch time in milliseconds when the token will expire.
+
+### Example PAT creation
+
+```http
+POST /api/users/123/personal-access-tokens HTTP/1.1
+Host: tenant.logto.app
+Content-Type: application/json
+Authorization: Bearer <access-token-with-management-api-access>
+
+{
+  "name": "My PAT",
+}
+```
+
+```http
+HTTP/1.1 201 Created
+Content-Type: application/json
+
+{
+  "name": "My PAT",
+  "value": "pat_W51arOqe7nynW75nWhvYogyc",
+  "expiresAt": null
+}
+```
+
+## Token Exchange with PAT
+
+Extend the token exchange flow to accept PATs as subject tokens and exchange them for short-lived access tokens.
+
+### Request
+
+The client makes a token exchange request to the tenant's token endpoint with a special grant type using the HTTP POST method. The following parameters are included in the HTTP request entity-body using the `application/x-www-form-urlencoded` format.
+
+1. `grant_type`: REQUIRED. The value of this parameter must be `urn:ietf:params:oauth:grant-type:token-exchange` indicates that a token exchange is being performed.
+2. `resource`: OPTIONAL. The resource indicator, the same as other token requests.
+3. `scope`: OPTIONAL. The requested scopes, the same as other token requests.
+4. `subject_token`: REQUIRED. The user's PAT.
+5. `subject_token_type`: REQUIRED. The type of the security token provided in the `subject_token` parameter. The value of this parameter must be `urn:logto:token-type:personal-access-token`.
+
+For security reasons, the client must authenticate with the token endpoint using the client ID and client secret in the `Authorization` header. Which means the client must be a traditional web application.
+
+And organization resources are not available for PATs.
+
+### Response
+
+If the token exchange request is successful, the tenant's token endpoint returns an access token that represents the identity of the user. The response includes the following parameters in the HTTP response entity-body using the `application/json` format.
+
+1. `access_token`: REQUIRED. The access token that represents the identity of the impersonated user.
```

Comment:
shall we emphasize that this token is the same token as other user-related grants return, e.g. `authorization_code` and `refresh_token`?

Reply:
updated